Task20

tzielin1 opened the pull request:
Function, documentation, and tests for regStats. Tests cover all 3 stats. Should we add text to the output to show how many zero ideals were extracted?

tzielin1 committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into 20-regStats

tzielin1 committed:
Added regStats to RandomMonomialIdeals.m2

tzielin1 committed:
Cleanup and filename comments to regStats

tzielin1 committed:
Added Documentation Outline

tzielin1 committed:
fixes

tzielin1 committed:
Fixes

So package installs, some temporary

tzielin1 committed:
Added Documentation

also changed instances of ShowRegularityTally to ShowTally for future update

sonjapetrovic committed:
Left a comment re: ShowRegularityTally !

sonjapetrovic committed:
small doc edits.

tzielin1 committed:
Merge branch 'TASK20' of https://github.com/RandCommAlg/RMI into TASK20

tzielin1 committed:
Cleanup from merge

tzielin1 committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into TASK20

# Conflicts:
#	RandomMonomialIdeals.m2

tzielin1 committed:
Cleanup from master merge

tzielin1 committed:
Reg of 0 ideal

tzielin1 committed:
fixes

tzielin1 committed:
fixed 2nd moment calc

tzielin1 committed:
Move avg + stdDev calcs

tzielin1 committed:
Added Tests and Fixed Documentation

sonjapetrovic committed:
added useful text after "end"

for copy+paste when testing package!

sonjapetrovic committed:
very minor change to doc.

Sondzus commented:
very minor change added to doc.
@tzielin1  I'm good with this branch - no issues right now. Perhaps for consistency someone else can double-check the branch before merging? 

LOL - update 3 days later - made some changes below.

Sondzus commented:
@tzielin1  I forgot to answer "Should we add text to the output to show how many zero ideals were extracted?" --- in my opinion, yes we should, as we do it for mingenStats as well. 
But see also new Task27 ... we need to decide what to print and what not to print jointly with @dkosmas and mingenStats.

In particular I'd suggest printing the number of 0 ideals extracted, since we are not returning that value; but I'd also suggest not printing the rest of the message (that is, sample size and average - why print those when they are easily computed and/or actually returned?). @dwilburne @sdgpfo  agree?

tzielin1 committed:
Added text outputs about # of zero ideals

tzielin1 commented:
I added text output about how many zero ideals were in the set, check it out

Sondzus commented:
@tzielin1 Great. I have 2 comments now:
1. Does "All ideals in this **list** are the zero ideal" sound better than "All ideals in this **set** are the zero ideal"? 
2. Does the message print if ShowTally => true? I think it does not because line 284 is after a "return" statement. Here is what I have: 

i4 : B=randomMonomialIdeals(3,3,0.0,1);
There are 1 ideals in this sample.
Of those, 1 were the zero ideal.
i5 : regStats(B)
All ideals in this set are the zero ideal
o5 = (-infinity, 0)
o5 : Sequence
i6 : regStats(B,ShowTally=>true)
o6 = (-infinity, 0, Tally{-infinity => 1})
o6 : Sequence

tzielin1 committed:
Wording + fixed print position for showtally

tzielin1 commented:
I like list better yes, changed it to that plus fixed the print position when ShowTally is true.

sonjapetrovic committed:
removed ";" from doc example

Let the user see the ideals, so they can tell if the output makes any
sense at all or not. (with suppressed output, they have to guess what B
is,  making it slightly more difficult to understand the function we
are showcasing.)
also changed N=2 (from 3) since it was giving 3 copies of the same
ideal anyway.

Sondzus commented:
Looks like it's doing the correct thing now. I made another small tweak in doc. I think I'm good with this branch now; someone else should help resolve these conflicts, and then I say we can merge! :)

parkerjoncus committed:
Merge branch 'master' into TASK20
